cmd/bootstrap/cmd/finalize.go

```diff
@@ -263,3 +331,40 @@ func mergeNodeInfos(internalNodes, partnerNodes []model.NodeInfo) []model.NodeIn
 
 	return nodes
 }
+
+// Validation utility methods ------------------------------------------------
+
+func validateNodeID(nodeID flow.Identifier) flow.Identifier {
+	if nodeID == flow.ZeroID {
+		log.Fatal().Msg("NodeID must not be zero")
+	}
+	return nodeID
+}
+
+func validateNetworkPubKey(key encodable.NetworkPubKey) encodable.NetworkPubKey {
+	if key.PublicKey == nil {
+		log.Fatal().Msg("NetworkPubKey must not be nil")
+	}
+	return key
+}
+
+func validateStakingPubKey(key encodable.StakingPubKey) encodable.StakingPubKey {
+	if key.PublicKey == nil {
```

Comment:
This validation is insufficient. Have to use [validateAddressFormat](https://github.com/onflow/flow-go/blob/e7d87c2a8e2fc3e1ef6613729c7ee7ff51467810/cmd/bootstrap/cmd/key.go#L104)